frontend/src/metabase/query_builder/components/filters/modals/BulkFilterSelect/BulkFilterSelect.tsx

```diff
@@ -79,9 +85,62 @@ const BulkFilterSelect = ({
   );
 };
 
-const getNewFilter = (query: StructuredQuery, dimension: Dimension) => {
+const getNewFilter = (query: StructuredQuery, dimension: Dimension): Filter => {
   const filter = new Filter([], null, dimension.query() ?? query);
   return filter.setDimension(dimension.mbql(), { useDefaultOperator: true });
 };
 
-export default BulkFilterSelect;
+export interface SegmentFilterSelectProps {
+  query: StructuredQuery;
+  filters?: Filter[];
+  segments: SegmentOption[];
+  onAddFilter: (filter: Filter) => void;
+  onRemoveFilter: (filter: Filter) => void;
+}
+
+export const SegmentFilterSelect = ({
+  query,
+  filters,
+  segments,
+  onAddFilter,
+  onRemoveFilter,
+}: SegmentFilterSelectProps): JSX.Element => {
+  const activeSegments = useMemo(() => {
+    return segments.filter(segment => {
+      return !!filters?.find(
+        filter => filter[0] === "segment" && filter[1] === segment.filter[1],
+      );
+    });
+  }, [filters, segments]);
+
+  const toggleSegment = useCallback(
+    (newActiveSegments: SegmentOption[]) => {
+      const [changedSegment] = xor(newActiveSegments, activeSegments);
+      const segmentIsActive = activeSegments.includes(changedSegment);
+
+      const segmentFilter = new Filter(changedSegment.filter, null, query);
+
+      segmentIsActive
+        ? onRemoveFilter(segmentFilter)
+        : onAddFilter(segmentFilter);
+    },
+    [query, activeSegments, onRemoveFilter, onAddFilter],
+  );
+
+  return (
+    <div>
+      <Select
```

Comment:
There is a small UI glitch when adding the first segment. For regular filters the issue is solved by setting the height of `Select`, but maybe there is another fix 

<img width="723" alt="Screenshot 2022-05-24 at 14 32 22" src="https://user-images.githubusercontent.com/8542534/170025263-c41b4088-0de6-4663-96e8-d93b7ff61e2e.png">
<img width="703" alt="Screenshot 2022-05-24 at 14 32 33" src="https://user-images.githubusercontent.com/8542534/170025270-5669f821-7399-45b0-870e-8ce90f6cecdf.png">